hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/utils/WeightValidationHelper.java

```diff
@@ -49,19 +49,33 @@ public static WeightValidationHelper usingContext(final SystemSecurityContext sy
 
     /**
      * Validating weights associated with all the {@link DeploymentRequest}s
-     * 
+     *
      * @param deploymentRequests
      *            the {@linkplain List} of {@link DeploymentRequest}s
      */
     public void validate(final List<DeploymentRequest> deploymentRequests) {
         final long assignmentsWithWeight = deploymentRequests.stream()
-                .filter(request -> request.getTargetWithActionType().getWeight() != null).count();
+                .filter(request -> request.getTargetWithActionType().getWeight().isPresent()).count();
         final boolean containsAssignmentWithWeight = assignmentsWithWeight > 0;
         final boolean containsAssignmentWithoutWeight = assignmentsWithWeight < deploymentRequests.size();
 
         validateWeight(containsAssignmentWithWeight, containsAssignmentWithoutWeight);
     }
 
+    /**
+     * Validating weights associated with all the {@link DeploymentRequest}s accepting null weight
+     *
+     * @param deploymentRequests
+     *            the {@linkplain List} of {@link DeploymentRequest}s
+     */
+    public void validateAcceptNullWeight(final List<DeploymentRequest> deploymentRequests) {
+        final long assignmentsWithWeight = deploymentRequests.stream()
```

Comment:
code seems duplicated from validate(List<DeploymentRequest>). what aboout two static methods containsAssignmentWithWeight(...) and containsAssignmentWithWeight (...)?